x-pack/plugins/security/server/lib/saved_objects_client/secure_options_builder.js

```diff
@@ -7,13 +7,13 @@
 /*! Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one or more contributor license agreements.
  * Licensed under the Elastic License; you may not use this file except in compliance with the Elastic License. */
 
-export function secureSavedObjectsClientOptionsBuilder(server, options) {
+export function secureSavedObjectsClientOptionsBuilder(server, requestHasPrivileges, options) {
```

Comment:
I was attempting to follow the convention that you established by passing it the `server` as a separate parameter. That way the `options` parameter is the unmodified options that would normally be passed to the `SavedObjectClient` and we're supplementing it with addition values all inside the `secureSavedObjectsClientOptionsBuilder`.

When implementing this, I began to question the benefit of the `secureSavedObjectsClientOptionsBuilder` function and at one point considered inlining the whole  thing, but decided on holding off on doing so since you wrote the initial implementation and saw the merit of having it a separate function.

Reply:
Well I can't fault you there then 😄

In a perfect world, we would not need the `secureSavedObjectsClientOptionsBuilder`, but we require this because the options that come out of these builders are passed to the base OSS SavedObjectsClient, and the security plugin needs to swap out the default `callCluster` implementation when RBAC is enabled (`callWithInternalUser` vs `callWithRequest`).

The SOC wrappers don't have the ability to modify the original set of `options`, which is where these builders come in to play.

Reply:
Gotcha, when I mentioned inline it earlier, I meant something similar to the following in the security/index.js:

```
      savedObjectsClientProvider.addClientOptionBuilder(options => {
        const adminCluster = server.plugins.elasticsearch.getCluster('admin');
        const { callWithInternalUser } = adminCluster;

        return {
          ...options,
          callCluster: callWithInternalUser,
          requestHasPrivileges
        };
      });
```

Reply:
I don't feel too strongly either way at this point, I think it all depends on how we surface the `hasPrivileges` check to other consumers, which I've put off deciding at this moment.